demo/BERT/builder.py

```diff
@@ -524,26 +524,26 @@ def load_onnx_weights_and_quant(path, config):
 
 def emb_layernorm(builder, network, config, weights_dict, builder_config, sequence_length, batch_sizes):
     if len(batch_sizes) > 1:
-        input_ids = network.add_input(name="input_ids", dtype=trt.int32, shape=(sequence_length, -1))
-        segment_ids = network.add_input(name="segment_ids", dtype=trt.int32, shape=(sequence_length, -1))
-        input_mask = network.add_input(name="input_mask", dtype=trt.int32, shape=(sequence_length, -1))
+        input_ids = network.add_input(name="input_ids", dtype=trt.int32, shape=(-1, sequence_length))
+        segment_ids = network.add_input(name="segment_ids", dtype=trt.int32, shape=(-1, sequence_length))
+        input_mask = network.add_input(name="input_mask", dtype=trt.int32, shape=(-1, sequence_length))
 
         # Specify profiles for the batch sizes we're interested in.
         # Make sure the profile also works for all sizes not covered by the previous profile.
         prev_size = 0
         for batch_size in sorted(batch_sizes):
             profile = builder.create_optimization_profile()
-            min_shape = (sequence_length, prev_size + 1)
-            shape = (sequence_length, batch_size)
-            profile.set_shape("input_ids", min=min_shape, opt=shape, max=shape)
-            profile.set_shape("segment_ids", min=min_shape, opt=shape, max=shape)
-            profile.set_shape("input_mask", min=min_shape, opt=shape, max=shape)
+            min_shape = (1, sequence_length)
+            shape = (batch_size, sequence_length)
+            max_shape = (batch_size + 1, sequence_length)
```

Comment:
Out of curious, why max_shape is `batch_size + 1` when user only request `batch_size` in the command line?

Reply:
Keeping min, max and opt shapes different, will ensure the batch dimension carries -1  flag for the dynamic shape

Reply:
not necessary, -1 is guaranteed when you call `network.add_input` with `-1` in the dimension input. Could you revert bac to batch_size to consistent with the helper of option `--batch-size`? thanks.

Reply:
Unfortunately, the `getBindingDimensions(int binding_index)` is profile specific. A dimension will not be marked with "-1" if the profile has max_shape==min_shape==opt_shape(https://github.com/NVIDIA/TensorRT/blob/master/include/NvInferRuntime.h#L1163-L1181). Triton 20.08 depends on the "-1" flag to support batching.

I would recommend: `max_shape = (batch_size, sequence_length)` instead of `max_shape = (batch_size + 1, sequence_length)`.
Then in the model's config.pbtxt add the following:
`instance_group { profile : ["0", "1" ] }`

We have fixed the Triton server to support optimization profiles with static shapes(max_shape==min_shape==opt_shape).

Reply:
@ttyio to avoid (max_shape==min_shape==opt_shape), I'm gonna leave it as is. please let me know if it causes any issues

Reply:
The best practice is that we have a piecewise function for multiple profile in dynamic shape. The new logic in this code might confuse the customer, e.g, in the original code if user specify `-b 8 -b 16`, he expected to get 2 profiles [min=1, max=8], [min=9, max=16]. And in your new code, user will get 2 profiles [min = 1, max=9], [min = 1, max = 17]. Could you polish your code again? thanks~

Reply:
for Triton, profiles need to start with min=1 always :( @tanmayv25 if we want to fix this in Triton, is it a big change? 
@ttyio If it is a big change in TRITON - can we go with this? if it's only a matter of syntax and clarity, I would rather make the changes in TRT builder.

Reply:
Maybe `-b 8 -b 16` can be parse as `[min=1, max=9]` `[min=9, max=16]`, @rajeevsrao what do you think? thanks~

Reply:
We have relaxed the min_shape requirement in master. The fix will be available in Triton 20.09.

Reply:
ok can we push this for now @ttyio ? Will resend an MR with Triton 20.09

Reply:
Hi @swethmandava , I had a syncup with @rajeevsrao , this change is for Triton only and may break exist user that already use demoBert in 2 ways:
-  The opt no longer equal to max with this change
-  If user select profile by checking the input shape within profile range, he may find multiple profile match the query condition.
And the conclusion here is to revert the profile shape changes in this MR, and update in the READMme tells user that the `-b 1` has known issue when run with Triton, we can remove the known issue after Triton 20.09. What do you think?

Reply:
Sure, pushed the fix. Let me know if you need anything else to merge.